### PR TITLE
Adding source Manga Mutiny

### PR DESCRIPTION
**Pull request by E3FxGaming**
Reason for source addition: source requested in issue #3292

What does work (tested it on a virtual device - I welcome any additional testers):
- Browse/Latest
- filtering (genre, status and type) / sorting
- reading manga
- manga details (author/artist/status/description/genres)
- downloading manga

Webview doesn't work. Reason: it's an HttpSource based on an undocumented API - it's not based on web scraping

Credit:
- the concept for the Filter was taken from the MangaPark extension and changed to fit the circumstances
- the JsonObject extension function stems from stackoverflow

The entire extension is based on an undocumented API which I discovered by analyzing the network traffic after using the website. The API is stable (I didn't encounter any unexpected behavior) and the data from the JSON files is way easier to process than any web scraping on that website (let's leave that at "this website uses few descriptive/useful css class names")

### CODE REVIEW

**Comment by SnakeDoc83** [comment]
Regarding JSON, you might want to look in to using Kotson. What you've got here is fine, but for future projects you might find it easier to work with when parsing JSON.

**Comment by E3FxGaming** [comment]
> you might want to look in to using Kotson

Ah ok, I'll keep that in mind. I have basic GSON knowledge from when I used it with Java and after switching to more and more Kotlin I made myself familiar with com.squareup.moshi:moshi - however that would be an additional dependency which I wanted to avoid. Kotson is already exposed to extensions through common.gradle - so that would definitely be an option.

I'm not familiar with Kotson yet and at some point one half of this extension still relied on terrible web scraping - then I noticed that I can realize the entire extension with the API. That's when my reliance on JSON grew, but the other half of the extension already used GSON so I stuck with that.

File: src/mangamutiny/src/eu/kanade/tachiyomi/extension/en/mangamutiny/MangaMutiny.kt [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+    val value: JsonElement = this.get(key) ?: return null
+
+    if (value.isJsonNull) {
+        return null
+    }
+
+    return value
+}
+
+open class MangaMutiny : HttpSource() {
```
Comment (SnakeDoc83) [inline src/mangamutiny/src/eu/kanade/tachiyomi/extension/en/mangamutiny/MangaMutiny.kt:34]: Nitpick: class doesn't need to be open.

File: src/mangamutiny/src/eu/kanade/tachiyomi/extension/en/mangamutiny/MangaMutiny.kt [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+            responseBody.close()
+        }
+
+        return MangasPage(mangasPage, mangasPage.size == 20)
+    }
+
+    private fun mangaRequest(page: Int, filters: FilterList? = null, query: String? = null): Request {
+        val uri = Uri.parse(baseUrl).buildUpon()
+        uri.appendEncodedPath(apiMangaUrlPath)
+        if ((query != null) && query.isNotEmpty()) {
```
Comment (SnakeDoc83) [inline src/mangamutiny/src/eu/kanade/tachiyomi/extension/en/mangamutiny/MangaMutiny.kt:229]: Could do `if (query?.isNotBlank() == true)`

**Comment by E3FxGaming** [comment]
> It seems like the folder structure is wrong, /mangamutiny/ should be in /src/en/

I apologize for that. I had to reset my fork before asking for a merge, so I ended up backing all Manga Mutiny files up into a separate directory, resetting the fork and then something seems to have gone wrong during the restoration of my backed up files.

Should be fixed with my "MangaMutiny - multiple fixes" commit.

**Pull request merged**